src/drivers/intel/alh.c

```diff
@@ -146,6 +146,11 @@ static int alh_get_fifo(struct dai *dai, int direction, int stream_id)
 	return ALH_BASE + offset + ALH_STREAM_OFFSET * stream_id;
 }
 
+static uint32_t alh_get_fifo_depth(struct dai *dai, int direction)
+{
+	return dai->plat_data.fifo[direction].depth;
```

Comment:
any specific reason why in some drivers we check invalid directions and in others we don't? Maybe that check could be moved to the `dai_get_fifo_depth()` wrapper if it's common?

Reply:
not sure about this. just tried to copy the existing functionality...